tests/unit_tests/sparse_blas/source/sparse_gemm_buffer.cpp

```diff
@@ -0,0 +1,233 @@
+    oneapi::mkl::layout col_major = oneapi::mkl::layout::col_major;
+    int nrows_A = 4, ncols_A = 6, ncols_C = 5;
+    int ldb = transpose_A == oneapi::mkl::transpose::nontrans ? ncols_A : nrows_A;
+    int ldc = transpose_A == oneapi::mkl::transpose::nontrans ? nrows_A : ncols_A;
+
+    // Basic test
+    EXPECT_TRUEORSKIP(test(dev, nrows_A, ncols_A, ncols_C, density_A_matrix, index_zero, col_major,
+                           transpose_A, transpose_B, fp_one, fp_zero, ldb, ldc));
+    // Test index_base 1
+    EXPECT_TRUEORSKIP(test(dev, nrows_A, ncols_A, ncols_C, density_A_matrix,
+                           oneapi::mkl::index_base::one, col_major, transpose_A, transpose_B,
+                           fp_one, fp_zero, ldb, ldc));
+    // Test non-default alpha
+    EXPECT_TRUEORSKIP(test(dev, nrows_A, ncols_A, ncols_C, density_A_matrix, index_zero, col_major,
+                           transpose_A, transpose_B, set_fp_value<fpType>()(2.f, 1.5f), fp_zero,
+                           ldb, ldc));
+    // Test non-default beta
+    EXPECT_TRUEORSKIP(test(dev, nrows_A, ncols_A, ncols_C, density_A_matrix, index_zero, col_major,
+                           transpose_A, transpose_B, fp_one, set_fp_value<fpType>()(3.2f, 1.f), ldb,
+                           ldc));
+    // Test 0 alpha
+    EXPECT_TRUEORSKIP(test(dev, nrows_A, ncols_A, ncols_C, density_A_matrix, index_zero, col_major,
+                           transpose_A, transpose_B, fp_zero, fp_one, ldb, ldc));
+    // Test 0 alpha and beta
+    EXPECT_TRUEORSKIP(test(dev, nrows_A, ncols_A, ncols_C, density_A_matrix, index_zero, col_major,
+                           transpose_A, transpose_B, fp_zero, fp_zero, ldb, ldc));
+    // Test non-default ldb
+    EXPECT_TRUEORSKIP(test(dev, nrows_A, ncols_A, ncols_C, density_A_matrix, index_zero, col_major,
+                           transpose_A, transpose_B, fp_one, fp_zero, ldb + 5, ldc));
+    // Test non-default ldc
+    EXPECT_TRUEORSKIP(test(dev, nrows_A, ncols_A, ncols_C, density_A_matrix, index_zero, col_major,
+                           transpose_A, transpose_B, fp_one, fp_zero, ldb, ldc + 6));
+    // Test row major layout
+    EXPECT_TRUEORSKIP(test(dev, nrows_A, ncols_A, ncols_C, density_A_matrix, index_zero,
+                           oneapi::mkl::layout::row_major, transpose_A, transpose_B, fp_one,
+                           fp_zero, ncols_C, ncols_C));
+    // Test int64 indices
+    long long_nrows_A = 27, long_ncols_A = 13, long_ncols_C = 6;
+    long long_ldb = transpose_A == oneapi::mkl::transpose::nontrans ? long_ncols_A : long_nrows_A;
+    long long_ldc = transpose_A == oneapi::mkl::transpose::nontrans ? long_nrows_A : long_ncols_A;
+    EXPECT_TRUEORSKIP(test(dev, long_nrows_A, long_ncols_A, long_ncols_C, density_A_matrix,
+                           index_zero, col_major, transpose_A, transpose_B, fp_one, fp_zero,
+                           long_ldb, long_ldc));
+}
```

Comment:
Do we plan to add tests with invalid input arguments in the future? I don't think that it should be done                                                                                                                                                                                                                                                                                                                                                 in this PR, I'm just curious :)

Reply:
That's a good point, this would be nice. I've created a task so we may be able to work on that in the future.

Reply:
great